File: Pythonleetcode/Leetcode509.py
class Solution(object):
    def fib(self, N):
        if N==0:
            return 0
        if N<=2:
            return 1
        F=[0]*N
        F[0]=1
        F[1] = 1
        for i in range(2,N):
            F[i]=F[i-1]+F[i-2]
        return F[-1]



print(Solution().fib(3))
print(Solution().fib(4))

File: CppPractice/Project1/Project1/Leetcode509.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

using namespace std;


class Solution {
public:
    int fib(int n) {
        vector<int> dp;
        dp.push_back(0);
        dp.push_back(1);
            
        if (n<=1)
        {
            return dp[n];
        }
        for (int i = 2; i < n+1; i++)
        {
            dp.push_back(dp[i - 2] + dp[i - 1]);
        }

        return dp.back();
    }
};